File: C# .Net/JDI UI Framework/Tests/TutorialTests/.Dummies/JDI/Dropdown.cs

using System;
using System.Collections.Generic;
using System.Reflection;
using Epam.JDI.Core.Attributes.Functions;
using Epam.JDI.Core.Interfaces.Base;
using Epam.JDI.Core.Interfaces.Complex;
using OpenQA.Selenium;

namespace Epam.Tests.TutorialTests.Dummies.JDI
{
    public class Dropdown : IDropDown
    {
        private By by1;
        private By by2;

        public Dropdown() { }

        public Dropdown(By by1, By by2)
        {
            this.by1 = by1;
            this.by2 = by2;
        }

        public IAvatar Avatar
        {
            get
            {
                throw new NotImplementedException();
            }

            set
            {
                throw new NotImplementedException();
            }
        }

        public string Name
        {
            get
            {
                throw new NotImplementedException();
            }

            set
            {
                throw new NotImplementedException();
            }
        }

        public object Parent
        {
            get
            {
                throw new NotImplementedException();
            }

            set
            {
                throw new NotImplementedException();
            }
        }

        public string ParentTypeName
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public string TypeName
        {
            get
            {
                throw new NotImplementedException();
            }

            set
            {
                throw new NotImplementedException();
            }
        }

        public void Click()
        {
            throw new NotImplementedException();
        }

        public void Close()
        {
            throw new NotImplementedException();
        }

        public void Expand()
        {
            throw new NotImplementedException();
        }

        public string GetAttribute(string name)
        {
            throw new NotImplementedException();
        }

        public List<string> GetNames()
        {
            throw new NotImplementedException();
        }

        public List<string> GetOptions()
        {
            throw new NotImplementedException();
        }

        public string GetOptionsAsText()
        {
            throw new NotImplementedException();
        }

        public string GetSelected()
        {
            throw new NotImplementedException();
        }

        public int GetSelectedIndex()
        {
            throw new NotImplementedException();
        }

        public string GetText()
        {
            throw new NotImplementedException();
        }

        public string GetValue()
        {
            throw new NotImplementedException();
        }

        public List<string> GetValues()
        {
            throw new NotImplementedException();
        }

        public bool IsDisplayed()
        {
            throw new NotImplementedException();
        }

        public bool IsHidden()
        {
            throw new NotImplementedException();
        }

        public bool IsSelected(IConvertible name)
        {
            throw new NotImplementedException();
        }

        public bool IsSelected(string name)
        {
            throw new NotImplementedException();
        }

        public void Select(int index)
        {
            throw new NotImplementedException();
        }

        public void Select(IConvertible name)
        {
            throw new NotImplementedException();
        }

        public void Select(string name)
        {
            throw new NotImplementedException();
        }

        public void SetAttribute(string attributeName, string value)
        {
            throw new NotImplementedException();
        }

        public void SetFunction(Functions function)
        {
            throw new NotImplementedException();
        }

        public void SetName(FieldInfo field)
        {
            throw new NotImplementedException();
        }

        public void SetValue(string value)
        {
            throw new NotImplementedException();
        }

        public void WaitAttribute(string name, string value)
        {
            throw new NotImplementedException();
        }

        public void WaitDisplayed()
        {
            throw new NotImplementedException();
        }

        public string WaitMatchText(string regEx)
        {
            throw new NotImplementedException();
        }

        public void WaitSelected(IConvertible name)
        {
            throw new NotImplementedException();
        }

        public void WaitSelected(string name)
        {
            throw new NotImplementedException();
        }

        public string WaitText(string text)
        {
            throw new NotImplementedException();
        }

        public void WaitVanished()
        {
            throw new NotImplementedException();
        }
    }
}

File: jdi-uitest-win/src/main/java/com/epam/jdi/uitests/win/winnium/elements/complex/Dropdown.java
package com.epam.jdi.uitests.win.winnium.elements.complex;

import com.epam.jdi.uitests.core.interfaces.complex.IDropDown;
import com.epam.jdi.uitests.win.winnium.elements.base.Element;
import com.epam.jdi.uitests.win.winnium.elements.base.managers.DropdownManager;
import org.openqa.selenium.By;

import java.util.List;
import java.util.function.Supplier;

public class Dropdown<TEnum extends Enum & Supplier<String>> extends Element implements IDropDown<TEnum> {
    private DropdownManager<TEnum> dropdownManager;

    @Override
    public void click() {
        dropdownManager.click();
    }

    @Override
    public void expand() {
        dropdownManager.expand();
    }

    @Override
    public void close() {
        dropdownManager.close();
    }
    @Override
    public void select(String name) {
        dropdownManager.select(name);
    }

    @Override
    public void select(TEnum name) {
        dropdownManager.select(name);
    }

    @Override
    public void select(int index) {
        dropdownManager.select(index);
    }

    @Override
    public String getSelected() {
        return dropdownManager.getSelected();
    }

    @Override
    public int getSelectedIndex() {
        return dropdownManager.getSelectedIndex();
    }

    @Override
    public boolean isSelected(String name) {
        return dropdownManager.isSelected(name);
    }

    @Override
    public boolean isSelected(TEnum name) {
        return dropdownManager.isSelected(name);
    }

    @Override
    public void waitSelected(String name) {
        dropdownManager.waitSelected(name);
    }

    @Override
    public void waitSelected(TEnum name) {
        dropdownManager.waitSelected(name);
    }

    @Override
    public List<String> getOptions() {
        return dropdownManager.getOptions();
    }

    @Override
    public void setValue(String value) {
        dropdownManager.setValue(value);
    }

    @Override
    public String getText() {
        return dropdownManager.getText();
    }

    @Override
    public String waitText(String text) {
        return dropdownManager.waitText(text);
    }

    @Override
    public String waitMatchText(String regEx) {
        return dropdownManager.waitMatchText(regEx);
    }

    @Override
    public String getValue() {
        return dropdownManager.getValue();
    }

    public void setUp(By expandBy, By elementsByName) {
        if (expandBy != null)
            avatar.setByLocator(expandBy);

        dropdownManager = new DropdownManager<>(this, elementsByName);
    }
}
